File: Java/Show/src/TheRayCode/Show/shapes/Rectangle.java
package TheRayCode.Show.shapes;

public class Rectangle extends Shape {
    public int width;
    public int height;
    public Rectangle() {
    }
    public Rectangle(Rectangle target) {
        super(target);
        if (target != null) {
            this.width = target.width;
            this.height = target.height;
        }
    }

    @Override
    public Shape clone() {
        return new Rectangle(this);
    }
    public boolean equals(Object object) {
        if (!(object instanceof Rectangle) || !super.equals(object)) return false;
        Rectangle shape = (Rectangle) object;
        return shape.width == width && shape.height == height;
    }
}

File: Language/CSharp/Structural/Bridge/Example3/Rectangle.cs
namespace Show
{
    public class Rectangle: Shape
    {
        private int width, height, x, y;
        
        public Rectangle(int width, int height, int x, int y, IDrawAPI drawAPI) : base(drawAPI)
        {
            this.width = width;
            this.height = height;
            this.x = x;
            this.y = y;
        }
        
        
        public Rectangle(IDrawAPI drawAPI) : base(drawAPI)
        {
        }

        public override void Draw()
        {
            drawAPI.DrawRectangle(width, height, x, y);
        }
    }
}